pkg/saga/statemachine/statelang/parser/compensation_trigger_state_parser.go

```diff
@@ -0,0 +1,31 @@
+package parser
+
+import (
+	"github.com/seata/seata-go/pkg/saga/statemachine/constant"
+	"github.com/seata/seata-go/pkg/saga/statemachine/statelang"
+	"github.com/seata/seata-go/pkg/saga/statemachine/statelang/state"
+)
+
+type CompensationTriggerStateParser struct {
+	BaseStateParser
```

Comment:
I found `BaseStateParser` are sometimes used as pointer embedding, and sometimes value embedding. Is there any special consideration?

Reply:
I have modified them all as pointer embedding.